client/src/main/scala/io/delta/sharing/spark/RemoteDeltaLog.scala

```diff
@@ -165,7 +169,9 @@ class RemoteSnapshot(
   }
 
   private def getTableMetadata: (Metadata, Protocol, Long) = {
-    val tableMetadata = client.getMetadata(table, versionAsOf, timestampAsOf)
+    val tableMetadata = initDeltaTableMetadata.getOrElse(
+      client.getMetadata(table, versionAsOf, timestampAsOf)
+    )
```

Comment:
Question: 
Should we cache the metadata as well?

We could have a local var that can be initialzed from initDeltaTableMetadata and update it if not set, which will avoid a refetch across multiple calls even if initDeltaTableMetadata is not specified.

Reply:
updated. using val as there's no re-assignment to it i the code.